plugins/aws/test/resources/cloudfront_test.py

```diff
@@ -0,0 +1,197 @@
+from typing import cast, Any
+from types import SimpleNamespace
+from test.resources import round_trip_for
+
+from resoto_plugin_aws.aws_client import AwsClient
+from resoto_plugin_aws.resource.cloudfront import (
+    AwsCloudFrontDistribution,
+    AwsCloudFrontFunction,
+    AwsCloudFrontPublicKey,
+    AwsCloudFrontRealtimeLogConfig,
+    AwsCloudFrontResponseHeadersPolicy,
+    AwsCloudFrontStreamingDistribution,
+    AwsCloudFrontOriginAccessControl,
+    AwsCloudFrontCachePolicy,
+    AwsCloudFrontFieldLevelEncryptionConfig,
+    AwsCloudFrontFieldLevelEncryptionProfile,
+)
+from test import aws_client, aws_config  # noqa: F401
+
+
+def test_distributions() -> None:
+    first, builder = round_trip_for(AwsCloudFrontDistribution)
+    assert len(builder.resources_of(AwsCloudFrontDistribution)) == 1
+    # assert len(first.tags) == 1
+
+
+def test_tagging() -> None:
+    dist, _ = round_trip_for(AwsCloudFrontDistribution)
+
+    def validate_update_args(**kwargs: Any) -> Any:
+        if kwargs["action"] == "tag-resource":
+            assert kwargs["Resource"] == dist.arn
+            assert kwargs["Tags"] == {"Items": [{"Key": "foo", "Value": "bar"}]}
+
+    def validate_delete_args(**kwargs: Any) -> Any:
+        if kwargs["action"] == "untag-resource":
+            assert kwargs["Resource"] == dist.arn
+            assert kwargs["TagKeys"] == {"Items": ["foo"]}
+
+    client = cast(AwsClient, SimpleNamespace(call=validate_update_args))
+    dist.update_resource_tag(client, "foo", "bar")
+
+    client = cast(AwsClient, SimpleNamespace(call=validate_delete_args))
+    dist.delete_resource_tag(client, "foo")
+
+
+# def test_distribution_deletion() -> None:
+#     dist, _ = round_trip_for(AwsCloudFrontDistribution)
+
+#     def validate_delete_args(**kwargs: Any) -> Any:
+#         assert kwargs["action"] == "delete-distribution"
+#         assert kwargs["Id"] == dist.id
+
+#     client = cast(AwsClient, SimpleNamespace(call=validate_delete_args))
+#     dist.delete_resource(client)
```

Comment:
on purpose? same for all other commented out test methods.

Reply:
uhh... this is still work in progress, so .. 😬